pkg/nitro_enclaves_cpu_plugin/device_plugin.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nitro_enclaves_cpu_plugin
+
+import (
+	"errors"
+	"fmt"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s-ne-device-plugin/pkg/config"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	deviceName                     = "nitro_enclaves_cpus"
+	devicePluginServerReadyTimeout = 10
+	deviceOfflineCPUsPath          = "/sys/devices/system/cpu/offline"
+)
+
+var cpuIdCounter = 0
+
+func (necdp *NitroEnclavesCPUDevicePlugin) socketPath() string {
```

Comment:
nit: may be slightly reshuffle it for readability?

`IBasicDevicePlugin` interface

`NitroEnclavesCPUDevicePlugin` struct

methods of `NitroEnclavesCPUDevicePlugin`